.eslintrc.js

```diff
@@ -1,8 +1,43 @@
 module.exports = {
-    "extends": "@microsoft/eslint-config-azuretools",
-    "rules": {
-        "import/no-internal-modules": [ "error", {
-            "allow": [ "antlr4ts/**" ]
-        } ]
-    }
+    env: {
+        es6: true,
+        node: true,
+    },
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+        project: 'tsconfig.json',
+        sourceType: 'module',
+    },
+    plugins: ['@typescript-eslint', 'import'],
+    extends: [
+        'eslint:recommended',
+        'plugin:@typescript-eslint/recommended',
+        'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    ],
+    rules: {
+        '@typescript-eslint/ban-types': 'off',
+        '@typescript-eslint/consistent-type-imports': 'error',
+        '@typescript-eslint/no-inferrable-types': 'off',
+        '@typescript-eslint/no-namespace': 'off',
+        '@typescript-eslint/no-unnecessary-type-assertion': 'off',
+        '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
+        '@typescript-eslint/prefer-regexp-exec': 'off',
+        '@typescript-eslint/require-await': 'off',
+        '@typescript-eslint/restrict-template-expressions': 'off',
+        '@typescript-eslint/unbound-method': 'off',
```

Comment:
I don't speak from experience here so I'm open to your feedback. My suggestion to turn this back on it's based on the research of this topic I did:

This rule ensures that class methods are always bound to the correct this context, avoiding runtime errors.

Example:
```jsx
class MyClass {
    name = 'Test';
    printName() {
        console.log(this.name);
    }
}

const myClass = new MyClass();
const printName = myClass.printName; // Unbound method
printName(); // Error: 'this' is undefined or refers to the wrong context
```

Solution with correct binding:
```jsx
const printName = myClass.printName.bind(myClass); // Properly bound method
printName(); // Outputs: "Test"
```

By turning this rule **on**, we'd ensure that other code contributors are prompted to bind methods properly, preventing hard-to-debug issues with `this` context.

Reply:
`        '@typescript-eslint/unbound-method': 'off',`

Reply:
Thanks for sharing, I wasn't aware of the fact that bind returning an `any`.